### PR TITLE
24 separate all classes into their own files

### PR DESCRIPTION
**Pull request by ArgusLi**

### CODE REVIEW

**Comment by ArgusLi** [comment]
### Testing Strategy

1. Uninstall dbt-core and dbt-dremio `using pip uninstall`.
2. Install dbt-dremio using `pip install -e [local directory]`.
3. Verify dbt-dremio is version 1.1.0 and dbt-core is 1.1.2.
4. Create new dbt project using `dbt-init class_testing` with Dremio software with u/p.
5. Verify `dbt debug` connects properly.
6. Add model that contains `SELECT * FROM Samples."samples.dremio.com"."zips.json" LIMIT 5` in the project/models directory.
7. Verify `dbt run` runs 3 models without errors.

**Comment by ArgusLi** [comment]
### Test Result

No errors generated and all verifications passed.

File: dbt/adapters/dremio/api/authentication.py [hunk 1]
```diff
@@ -67,3 +72,24 @@ def get_headers(self):
...
         return self.build_headers(authorization_field)
+
+@dataclass
+class Parameters:
+    base_url: str
+    authentication: DremioAuthentication    
+    is_cloud: boolean = True
+    cloud_project_id: Optional[str] = None
+
+def login(api_parameters: Parameters, timeout=10, verify=True):
```
Comment (ravjotbrar) [inline dbt/adapters/dremio/api/authentication.py:83]: Should login just be under endpoints.py with all the other endpoints?

Comment (ArgusLi) [reply dbt/adapters/dremio/api/authentication.py:83]: I feel as though login is separate enough from the other endpoints that it belongs in authentication. However, this is just personal preference, I can also see what you mean in having all endpoint functions being in endpoints.py. Would love to hear your input @jlarue26.

Comment (ArgusLi) [reply dbt/adapters/dremio/api/authentication.py]: Moved login into endpoints.py.

**Review by ravjotbrar — APPROVED** [review]

**Comment by jlarue26** [comment]
@ArgusLi @ravjotbrar what are your thoughts on moving endpoints, error, url_builder into a new package: `api/rest`? This "hides" them even further as no other package should be using them other than the `api` modules.

**Comment by ArgusLi** [comment]
I think that's a pretty good idea.

**Pull request merged**